crates/stellar-registry-cli/src/commands/install.rs

```diff
@@ -33,70 +35,135 @@ pub enum Error {
     Io(#[from] std::io::Error),
     #[error(transparent)]
     Strkey(#[from] stellar_strkey::DecodeError),
+    #[error(transparent)]
+    Config(#[from] stellar_cli::config::locator::Error),
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
-        let contract_id = testnet::contract_id();
-        let network = testnet::network();
-        let out_dir = self.out_dir.as_ref().unwrap();
-        let mut out_file = out_dir.join(&self.deployed_name);
+        // Use the network config from flattened args
+        let network_passphrase = self
+            .config
+            .network
+            .network_passphrase
+            .clone()
+            .expect("Network passphrase required");
+
+        let contract_id = self.get_contract_id().await?;
+        let contract = Contract::from_string(&contract_id)?;
+
+        self.config.locator.save_contract_id(
+            &network_passphrase,
+            &contract,
+            &self.contract_name,
+        )?;
+
+        // Fetch and save the WASM file
+        let mut out_file = self.out_dir.join(&self.contract_name);
         out_file.set_extension("wasm");
-        let id_file = out_file.parent().unwrap().join("contract_id.txt");
+
         let fetch_cmd = fetch::Cmd {
-            contract_id,
-            out_file: Some(out_file),
-            network,
+            contract_id: UnresolvedContract::Resolved(contract),
+            out_file: Some(out_file.clone()),
+            network: self.config.network.clone(),
             ..Default::default()
         };
         fetch_cmd.run().await?;
-        std::fs::write(id_file, testnet::contract_id_strkey().to_string())?;
+        println!("WASM file saved to: {}", out_file.display());
+
+        println!("✅ Successfully installed contract {}", self.contract_name);
+        println!("Contract ID: {contract_id}");
+
         Ok(())
     }
 
-    pub async fn get_contract_id(
-        &self,
-        contract_id: &UnresolvedContract,
-        network: &network::Args,
-    ) -> Result<String, Error> {
-        let mut cmd = invoke::Cmd {
-            contract_id: contract_id.clone(),
-            config: config::Args {
-                network: network.clone(),
-                ..Default::default()
-            },
-            is_view: true,
-            ..Default::default()
-        };
-        cmd.slop = vec!["fetch_contract_id", "--deployed_name", &self.deployed_name]
-            .into_iter()
-            .map(Into::into)
-            .collect::<Vec<_>>();
-        println!("Fetching contract ID...\n{cmd:#?}");
-        Ok(cmd
-            .invoke(&global::Args::default())
-            .await?
-            .into_result()
-            .unwrap())
+    pub async fn get_contract_id(&self) -> Result<String, Error> {
```

Comment:
Perhaps return `stellar_strkey::Contract` here. Also you can use `Ok(contract.parse()?)`